pytorch_lightning/loggers/wandb.py

```diff
@@ -213,8 +219,44 @@ def version(self) -> Optional[str]:
         # don't create an experiment if we don't have one
         return self._experiment.id if self._experiment else self._id
 
+    @rank_zero_only
```

Comment:
can we check this condition inside the trainer? That way it's easer to ensure that this was only called by rank 0. Otherwise this is an easy point of failure for new logger implementations

Reply:
Is it something that should be enforced?

I'm not sure of the logic but currently:
* the callback calls `ModelCheckpoint.save_checkpoint` which must run on all ranks and delegate proper responsibility to `self.save_function` (per docstring)
  * indirectly it calls `CheckpointConnector.save_checkpoint` which runs on rank 0
  * it also calls `logger.after_save_checkpoint` which does the same and run only on rank 0 (but maybe another logger would want something else?)

Reply:
I agree with @justusschock. Better to call after_save_checkpoint only on rank 0 and put this logic in ModelCheckpoint. If a use case requires all ranks, we will update the code then.

Reply:
Sounds good. I made the change.